### PR TITLE
[2.1] Revert "[Datasets] Bundle blocks smaller than batch size in `map_batches` tasks. (#28648)"

### PR DESCRIPTION
**Pull request by c21**
This reverts commit b5687a1aaa3dc72dfde9758e730607a4ab22afe4.

## Why are these changes needed?

To avoid benchmark regression in release 2.1.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/examples/torch_image_batch_pretrained.py [hunk 1]
```diff
@@ -36,4 +36,4 @@ def preprocess(batch: np.ndarray) -> pd.DataFrame:
 ckpt = TorchCheckpoint.from_model(model=model, preprocessor=preprocessor)
 
 predictor = BatchPredictor.from_checkpoint(ckpt, TorchPredictor)
-predictor.predict(dataset, batch_size=80)
+predictor.predict(dataset)
```
Comment (c21) [inline doc/source/ray-air/examples/torch_image_batch_pretrained.py:39]: This is manual change after resolving conflict, please double check @clarkzinzow.

File: python/ray/data/dataset.py [hunk 2]
```diff
@@ -516,42 +504,46 @@ def map_batches(
...
                     raise ValueError(
                         "The map batches UDF returned the value "
-                        f"{batch} of type {type(batch)}, "
+                        f"{applied} of type {type(applied)}, "
                         "which is not allowed. "
                         f"The return type must be one of: {BatchType}"
                     )
-                # Add output batch to output buffer.
-                output_buffer.add_batch(batch)
+                output_buffer.add_batch(applied)
```
Comment (c21) [inline python/ray/data/dataset.py:546]: This is manual change after resolving conflict, please double check @clarkzinzow.

**Review by clarkzinzow — APPROVED** [review]
LGTM

**Review by jianoaix — APPROVED** [review]

**Comment by c21** [comment]
The failed dataset test is due to "polars not installed".
Merged to release branch.

**Pull request merged**